xonsh/environ.py

```diff
@@ -891,6 +912,7 @@ def current_branch(pad=True):
         branch = get_hg_branch()
     if isinstance(branch, subprocess.TimeoutExpired):
         branch = '<branch-timeout>'
+        _first_branch_timeout_message()
```

Comment:
Not without doing something wacky. Branch color and branch name are currently fully independent and I think that is how they should remain.  That said, I do think we should have a combined `{color_and_branch}` and migrate the defaults to that. That function would then also call the message function on timeout.